src/main/java/seedu/address/logic/commands/NoteListCommand.java

```diff
@@ -1,42 +1,61 @@
 package seedu.address.logic.commands;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 import seedu.address.logic.CommandHistory;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
-import seedu.address.model.StorageController;
 import seedu.address.model.note.Note;
 import seedu.address.model.note.NoteManager;
 
+
 /**
  * Lists notes based on given predicate.
  */
 public class NoteListCommand extends Command {
 
     public static final String COMMAND_WORD = "note list";
 
-    public static final String MESSAGE_SUCCESS = "Listed %1$s note(s).";
+    public static final String MESSAGE_SUCCESS = "Listed %1$s note(s).\n";
 
     public static final String MESSAGE_NOT_FOUND = "No notes were found.";
 
-    private NoteManager noteManager = new NoteManager();
+    private final String moduleCode;
+
+    public NoteListCommand(String moduleCode) {
+        this.moduleCode = moduleCode;
+    }
 
     @Override
     public CommandResult execute(Model model, CommandHistory history) throws CommandException {
 
+        NoteManager noteManager = new NoteManager();
+        List<Note> filteredNotes;
+
+        if (moduleCode.trim().length() > 0) {
+            filteredNotes = noteManager.getNotes().stream()
+                    .filter(p -> p.getModuleCode().equalsIgnoreCase(moduleCode)).collect(Collectors.toList());
+        } else {
```

Comment:
Suggestion: you can extract this into a method in your `NoteManager` i.e. `getAllNotesByModuleCode`